tests/net/mgmt/src/mgmt.c

```diff
@@ -423,4 +423,32 @@ ZTEST(mgmt_fn_test_suite, test_mgmt)
 		      "test_synchronous_event_listener failed");
 }
 
+K_SEM_DEFINE(wait_for_event_processing, 0, 1);
+static void net_mgmt_event_handler(struct net_mgmt_event_callback *cb,
+				    uint32_t mgmt_event, struct net_if *iface)
+{
+	ARG_UNUSED(cb);
+	ARG_UNUSED(iface);
+	ARG_UNUSED(mgmt_event);
+
+	k_sem_give(&wait_for_event_processing);
+}
+
+ZTEST(mgmt_fn_test_suite, test_mgmt_duplicate_handler)
+{
+	struct net_mgmt_event_callback cb;
+	int ret;
+
+	net_mgmt_init_event_callback(&cb, net_mgmt_event_handler, 0);
+	net_mgmt_add_event_callback(&cb);
+	net_mgmt_init_event_callback(&cb, net_mgmt_event_handler, 0);
+	net_mgmt_add_event_callback(&cb);
+
+	ret = k_sem_take(&wait_for_event_processing, K_SECONDS(1));
+	zassert_equal(ret, -EAGAIN, "Event is not processed");
```

Comment:
We need to make this properly, we are still waiting one second as the handler is never called. My idea was that we send an event (via system workq with a delay) before taking the semaphore, then wait until the handler releases the semaphore, all this probably takes milliseconds. The call count variable then makes sure that the callback is called exactly once regardless of how many callback handlers we have installed. Installing more than one callback is not possible but we can certainly verify that here.

Reply:
We can directly call `net_mgmt_event_notify` as it just queues to the net_mgmt queue instead of system workq?

Reply:
Yes, that should work too. I was worried about the scheduling, but it does not matter here as if the semaphore is released by the handler function before the `k_sem_take`, then we just do not wait there.

Reply:
But `NET_MGMT_EVENT_THREAD` isn't enabled because of unsatisfied dependencies, there is a Kconfig warning too:

```
warning: NET_MGMT_EVENT_STACK_SIZE (defined at drivers/wifi/esp32/Kconfig.esp32:31,
subsys/net/ip/Kconfig.mgmt:58) was assigned the value '1024' but got the value ''. Check these
unsatisfied dependencies: ((WIFI_ESP32 && WIFI) || (NET_MGMT_EVENT_THREAD && NET_MGMT_EVENT &&
NETWORKING)) (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_NET_MGMT_EVENT_STACK_SIZE and/or look up
NET_MGMT_EVENT_STACK_SIZE in the menuconfig/guiconfig interface. The Application Development Primer,
Setting Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be
helpful too.
```

Reply:
Stopping the `thrower_thread` before the 2nd test seemed to work fine.

Reply:
Two fixes:

1. Deleted the event callback in second test - passed net.workqueue
2. Increased the stack size for thrower thread - passed net.synchronouse